tribe-ext-hide-additional-fields.php

```diff
@@ -86,21 +103,30 @@ public function override_additional_fields_template( $file ) {
 		}
 
 		public function filter_additional_fields( $data ) {
-			$hidden_fields = get_post_meta( get_the_ID(), 'custom-hidden', true );
+			$hidden_fields     = get_post_meta( get_the_ID(), self::$field_key, true );
 			$additional_fields = tribe_get_option( 'custom-fields', false );
-			$labels = wp_list_pluck( $additional_fields, 'label', 'name' );
-
-		    if ( tribe_is_event() && $hidden_fields ) {
-		       if ( is_array( $hidden_fields ) ) {
-					foreach ( $hidden_fields as $field ) {
-						$field_label = $labels[ $field ];
-						unset( $data[ $field_label ] );
-					}
-		        	return apply_filters( 'tribe_filter_hidden_fields', $data );
-				} else {
-					return apply_filters( 'tribe_filter_hidden_fields', $data );
+			$labels            = wp_list_pluck( $additional_fields, 'label', 'name' );
+
+			if ( '' === $hidden_fields ) {
+				// Keep database clean.
+				delete_post_meta( get_the_ID(), self::$field_key );
+			} elseif (
+				tribe_is_event()
+				&& is_array( $hidden_fields )
+				&& ! empty( $hidden_fields )
+			) {
+				foreach ( $hidden_fields as $field ) {
+					// Exact match.
+					$field_label = $labels[ $field ];
+					unset( $data[ $field_label ] );
+
+					// Handle $data having HTML entities but $additional_fields being raw value, such as `&#039;` vs `'`.
+					$field_label = esc_html( $field_label );
+					unset( $data[ $field_label ] );
 				}
-		    }
+			}
+
+			return apply_filters( 'tribe_filter_hidden_fields', $data );
```

Comment:
This filter needs to be better namespaced like: `tribe_extension_` + a docblock.

Reply:
done. please re-review.